aws/rds_cluster_snapshots_test.go

```diff
@@ -0,0 +1,169 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// Custom waiter function waitUntilRdsClusterAvailable uses the Amazon RDS API operation DescribeDBClustersInput
+// to wait for the cluster to be available before returning.
+// If the condition is not met within the max attempt window, an error will be returned.
+func waitUntilRdsClusterAvailable(svc *rds.RDS, name *string) error {
+	input := &rds.DescribeDBClustersInput{
+		DBClusterIdentifier: name,
+	}
+
+	for i := 0; i < 90; i++ {
+		_, err := svc.DescribeDBClusters(input)
+		if err != nil {
+			return err
+		}
+
+		time.Sleep(1 * time.Second)
+		logging.Logger.Debug("Waiting for RDS DB Cluster to be created")
+	}
+
+	return RdsClusterAvailableError{name: *name}
+}
+
+// Built-in waiter function WaitUntilDBClusterSnapshotAvailable not working as expected.
+// Created a custom one
+func waitUntilRdsClusterSnapshotAvailable(svc *rds.RDS, clusterName *string, snapshotName *string) error {
+	input := &rds.DescribeDBClusterSnapshotsInput{
+		DBClusterIdentifier:         clusterName,
+		DBClusterSnapshotIdentifier: snapshotName,
+	}
+	for i := 0; i < 90; i++ {
+		_, err := svc.DescribeDBClusterSnapshots(input)
+		if err != nil {
+			return err
+		}
+
+		time.Sleep(1 * time.Second)
+		logging.Logger.Debug("Waiting for RDS DB Cluster Snapshot to be created")
+	}
+
+	return RdsClusterSnapshotAvailableError{clusterName: *clusterName, snapshotName: *snapshotName}
+}
+
+// createTestDBCluster generates a test DB Cluster since snapshots can only be created
+// from an existing DB Cluster
+func createTestDBCluster(t *testing.T, session *session.Session, name string) {
+	svc := rds.New(session)
+	input := &rds.CreateDBClusterInput{
+		DBClusterIdentifier: awsgo.String(name),
+		Engine:              awsgo.String("aurora-mysql"),
+		MasterUsername:      awsgo.String("gruntwork"),
+		MasterUserPassword:  awsgo.String("password"),
+	}
+
+	_, err := svc.CreateDBCluster(input)
+	require.NoError(t, err)
+	waitUntilRdsClusterAvailable(svc, &name)
+
+}
+
+// createTestRDSClusterSnapshot generates a test DB Snapshot
+func createTestRDSClusterSnapshot(t *testing.T, session *session.Session, clusterName string, snapshotName string) {
+	svc := rds.New(session)
+	input := &rds.CreateDBClusterSnapshotInput{
+		DBClusterIdentifier:         awsgo.String(clusterName),
+		DBClusterSnapshotIdentifier: awsgo.String(snapshotName),
+	}
+
+	_, err := svc.CreateDBClusterSnapshot(input)
+	require.NoError(t, errors.WithStackTrace(err))
+	waitUntilRdsClusterSnapshotAvailable(svc, &clusterName, &snapshotName)
+
+	result, err := svc.DescribeDBClusterSnapshots(&rds.DescribeDBClusterSnapshotsInput{
+		DBClusterSnapshotIdentifier: &snapshotName,
+	})
+
+	// Generate a tag
+	var tags = []*rds.Tag{
+		{
+			Key:   awsgo.String("snapshot-tag1-key"),
+			Value: awsgo.String("snapshot-tag1-value"),
+		},
+	}
+
+	// Tag a DB Cluster snapshot resource
+	for _, database := range result.DBClusterSnapshots {
```

Comment:
Oh, wait, this tags _all_ the snapshots for the DB... Is that intentional? If it is, it should probably be extracted into a separate method that makes it clear that'll happen.

Reply:
Fixed